portfolio/src/main/webapp/markermap.js

```diff
@@ -125,8 +125,11 @@ class MarkerMap {
     fetch('/markers')
       .then(response => response.json())
       .then((markers) => {
-        for(let marker of this.markers) {
+        for (let marker of this.markers) {
```

Comment:
You should **never** modify the length of an array (remove elements, add elements, etc) while looping over it! It causes buggy behavior and is dependent on how languages implement arrays and loops. 
For example, here's what happens in JavaScript if you tried to remove elements from the loop while looping over it:
![image](https://user-images.githubusercontent.com/10354630/84692376-9bf80080-aefa-11ea-9bcd-7e873e876303.png)
-- it ends up removing every other element!
If you tried to append to a loop while for-of looping over it, you'd end up with an infinite loop because the for-of loop would never reach the last element (since the array is always growing). 

Since you are just trying to delete all the elements from `this.markers`, you can simply set `this.markers = []` after the loop.